tests/src/Test/Tasty/Clash.hs

```diff
@@ -407,63 +336,39 @@ runTest1
   -> TestTree
 runTest1 modName opts@TestOptions{..} path target =
   withResource mkTmpDir Directory.removeDirectoryRecursive $ \tmpDir ->
-    testGroup (show target) $
-      sequenceIndependent (clashTest tmpDir) $
-           (case verificationTool of
-             Nothing -> []
-             Just SymbiYosys -> sbyTests opts tmpDir)
-         : hdlTests tmpDir
+    testGroup (show target) $ pure $
+      afterTree
+        AllSucceed
+        (clashTest tmpDir)
+        (testGroup "external tools" (verifTests tmpDir : hdlTests tmpDir))
  where
   mkTmpDir = flip createTempDirectory "clash-test" =<< getCanonicalTemporaryDirectory
   sourceDir = List.foldl' (</>) sourceDirectory (reverse (tail path))
 
   clashTest tmpDir =
-    ("clash (gen)", singleTest "clash (gen)" (ClashGenTest {
+    singleTest "clash (gen)" (ClashGenTest {
       cgExpectFailure=expectClashFail
     , cgBuildTarget=target
     , cgSourceDirectory=sourceDir
     , cgExtraArgs=clashFlags
     , cgModName=modName
     , cgOutputDirectory=tmpDir
-    }))
-
-  buildAndSimTests sim (buildTests, simTests)
-    | isJust expectClashFail = []
-    | otherwise              =
-      (if sim `elem` hdlLoad then buildTests else []) <>
-      (if sim `elem` hdlSim then simTests else [])
-
-  -- HACK: We want to run verilator and simulator tests independently if they
-  -- are both going to be run, otherwise failures from whichever comes first
-  -- will mean the second is skipped. In lieu of a better way to sequence tests
-  -- we can sequence tests for multiple simulators, then drop the first test
-  -- tree for all but the first simulator (as it will be copies of clash gen).
-  --
-  -- TODO: Since tasty doesn't provide one, we should really provide a better
-  -- set of combinators for describing test dependencies. That way we can have
-  -- some more principled way of having a test structure like
-  --
-  --   Group A
-  --     - Task A1
-  --     - Group B
-  --         - Task B1
-  --         - Task B2
-  --     - Group C
-  --         - Task C1
-  --
-  -- where groups B and C are independent of each other, but both dependent on
-  -- the success of Task A1.
-  sequenceIndependent
-    :: (TestName, TestTree)
-    -- ^ The tree everything else depends on ("clash (gen)")
-    -> [[(TestName, TestTree)]]
-    -- ^ All the independent trees to run after "clash (gen)"
-    -> [TestTree]
-  sequenceIndependent gen = go
+    })
+
+  emptyGroup = testGroup "empty" []
+
+  buildAndSimTests sim (buildTests, simTests0)
+    | isJust expectClashFail = emptyGroup
+    | sim `notElem` hdlLoad  = emptyGroup
```

Comment:
There's a very slight change in behavior here: if a simulator isn't in `hdlLoad`, simulation also won't run. Is that fair @DigitalBrains1, or is there some subtlety I'm overlooking?

Reply:
Yeah, it's fair. It used to be (well for the past 10 days :-P ) that the Vivado tests wholly ignored the value of `hdlLoad`, but I still made sure `tests/Main.hs` correctly set `hdlLoad` to include Vivado or not. Since you also edited the Haddock for `vivadoTests`, all is well.